File: app/src/main/java/me/ykrank/s1next/viewmodel/BlackListViewModel.java
package me.ykrank.s1next.viewmodel;

import android.view.View;
import android.widget.TextView;

import androidx.databinding.ObservableField;

import com.google.android.material.snackbar.Snackbar;

import me.ykrank.s1next.data.db.dbmodel.BlackList;

import static com.github.ykrank.androidtools.ui.internal.CoordinatorLayoutAnchorDelegateBaseImpl.SNACK_BAR_MAX_LINE;


public final class BlackListViewModel {

    public final ObservableField<BlackList> blacklist = new ObservableField<>();

    public View.OnClickListener clickSnackbar() {
        return v -> {
            Snackbar snackbar = Snackbar.make(v.getRootView(), ((TextView) v).getText(), Snackbar.LENGTH_SHORT);
            TextView textView = snackbar.getView().findViewById(com.google.android.material.R.id.snackbar_text);
            textView.setMaxLines(SNACK_BAR_MAX_LINE);
            snackbar.show();
        };
    }
}

File: app/src/main/java/me/ykrank/s1next/viewmodel/BlackListViewModel.kt
package me.ykrank.s1next.viewmodel;

import android.view.View
import android.widget.TextView
import androidx.databinding.ObservableField
import com.github.ykrank.androidtools.ui.internal.CoordinatorLayoutAnchorDelegateBaseImpl
import com.google.android.material.R
import com.google.android.material.snackbar.Snackbar
import me.ykrank.s1next.data.db.dbmodel.BlackList


class BlackListViewModel {
    val blacklist: ObservableField<BlackList> = ObservableField()

    fun clickSnackbar(): View.OnClickListener {
        return View.OnClickListener { v: View ->
            val snackbar = Snackbar.make(
                v.rootView,
                (v as TextView).text,
                Snackbar.LENGTH_SHORT
            )
            val textView = snackbar.view
                .findViewById<TextView>(R.id.snackbar_text)
            textView.maxLines = CoordinatorLayoutAnchorDelegateBaseImpl.SNACK_BAR_MAX_LINE
            snackbar.show()
        }
    }
}
